library/shlib.pl

```diff
@@ -573,15 +573,16 @@
 % Under MSYS2, the program is invoked from a bash, with the dll
 % dependencies (zlib1.dll etc.) in %MINGW_PREFIX%/bin instead of
 % the installation directory). Here we add this folder to the dll
-% search path. If MINGW_PREFIX is undefined, the directive is
-% skipped.
+% search path.
 
 add_mingw_dll_directory :-
-    (   getenv('MINGW_PREFIX', Prefix)
-    ->  format(atom(Bin), '~w/bin', [Prefix]),
-        win_add_dll_directory(Bin)
-    ;   true
-    ).
+    current_prolog_flag(msys2, true),
+    !,
+    getenv('MINGW_PREFIX', Prefix),
```

Comment:
What if `MINGW_PREFIX` is not defined?

Reply:
This will raise an error message, that an initialization directive has failed. But that is ok, the msys2 version only works in an msys2 environment, so if the directive fails, there is a real problem.

Reply:
Is this actually necessary?   If dlls like libgmp, gcc runtime, etc are not in the search path, the system will not start at all, so we never get here?    When would we get here?

Reply:
Actually, it does start normally. The problem appears if I use_module(library(archive)), complaining about a missing module, that turns out to be zlib1.dll